File: verve patch/base/Applications/CSymphony_0_1/vbSamples/CodeTypeReferenceExpression.vb
Option Strict Off
Option Explicit On

Imports System
Imports System.CodeDom
Imports System.CodeDom.Compiler
Imports System.Collections
Imports CSymphony.Lib

Namespace CSymphony.CSy
	Public Class CSyUnit
		 Inherits GraphUnit
		Overrides Public Sub Build()
			ns("MyNameSpace")

			declClass(TypePublic, "MyClass")
			Dim typeDateTime As TypeRef = New TypeRef("System.DateTime")

			declMethod(MemPublic, "System.Void", "MyMethod")
			
			enter (typeDateTime.Expr())
		End Sub
	End Class
End Namespace
			

File: verve patch/base/Applications/CSymphony_0_1/vbSamples/SourceOut/CodeTypeReferenceExpression.cpp
#pragma once 

#using <mscorlib.dll>

namespace MyNameSpace
{
    
    using namespace System;
    public __gc class MyClass;
    
    
    public __gc class MyClass
    {
        
        public: virtual System::Void MyMethod();
    };
    
    
    inline System::Void MyClass::MyMethod()
    {
        System::DateTime;
    };
}
